### ISSUE TITLE
Idea: Click Buffering

### ISSUE THREAD

**Issue by GTrista** [lossycrypt/eradicators-hand-crank-generator#3]
I've noticed an annoyance of mistiming my presses, causing me to not crank the generator and either having to click again or just missing it completely.
Extremely annoying when you try to crank the generator to the beat of some music.

And I don't want to simply decrease the cooldown because I want to limit myself to a maximum power generation, so that I wouldn't just start using these as my primary power source for any longer but the early game.

Sadly, I have not enough experience in modifying mods in Factorio myself, so I ask you to add button buffering, that remembers, if it has been pressed once recently and cranks the generator when the cooldown is over.
Maybe even making it a checkmark on the start-up settings and a setting to say how many frames the buffer checks so you can choose if you want a buffer press so mistiming the generator isn't as annoying.

Also, sorry: I don't know how to change the topic type to not be an "Issue"

**Comment by GTrista** [comment]
I have learned how to mod using your mod... kinda lame that I didn't realize until now that your mod is also a glorified modding tutorial.

Anyways- I modified your mod, but the license you chose doesn't allow me to publish your mod with modifications- so instead I bring the modified mod to you if you are okay with me adding the mod to the Factorio modding hub. I didn't change the info.json in it yet, but I would mention it to be a fork of your mod.

The following changes I made:
- Made a charging animation, where it spins clockwise (only works while not currently discharging sadly)
- Changed the settings data, removing the amount of seconds per cranking into an amount of maximum power per cranking.
- Removed the Cranking interval
- Made the power given per cranking be dependant on when the last cranking happened (clamped at 60 frames - 1 second)
- Made the auto-cranker trigger at a fixed rate of every 60 frames (1 second)
- Added floating flavour text for when you haven't used the HCG for more than 5 seconds

So in conclusion, I managed to add the feature I requested for myself, but I can't apply it.

[eradicators-hand-crank-generator_2.0.10.zip](https://github.com/user-attachments/files/16731961/eradicators-hand-crank-generator_2.0.10.zip)

**Comment by lossycrypt** [comment]
I'm sorry if this answer is too late for you. I've not been active in the Factorio community for years. I did not expect this mod to generate interest for this many years and I'm very happy it has helped you get into modding. Thank you very much for respecting the license. You have convinced me to put the mod into the public domain where it can hopefully still fulfill its role as a tutorial. The mod portal page has been updated accordingly, but i don't have time to update the actual files. Another fan has already posted a derivative on the mod portal, maybe you can work together to keep clutter on the portal at a minimum. The factory must grow!
